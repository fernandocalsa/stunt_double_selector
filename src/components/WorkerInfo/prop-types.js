import PropTypes from 'prop-types'

export default {
  imageUrl: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired
}
