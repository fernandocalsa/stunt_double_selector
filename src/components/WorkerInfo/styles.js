export default {
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    border: '#000 solid 1px',
    borderRadius: 10,
    width: '100%',
    padding: '5px 20px'
  },
  image: {
    width: '100%'
  },
  name: {
    fontFamily: 'Tahoma, Geneva, sans-serif',
    fontSize: 20
  }
}