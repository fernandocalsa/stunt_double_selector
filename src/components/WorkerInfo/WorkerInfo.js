import React from 'react'
import styles from './styles'
import propTypes from './prop-types'

const WorkerInfo = (props) => {
  const { imageUrl, firstName, lastName } = props
  return (
    <div style={styles.container}>
      <img
        style={styles.image}
        src={imageUrl}
        alt={`${firstName} ${lastName}`}
        />
      <span style={styles.name}>{firstName} {lastName}</span>
    </div>
  )
}

WorkerInfo.propTypes = propTypes

export default WorkerInfo
