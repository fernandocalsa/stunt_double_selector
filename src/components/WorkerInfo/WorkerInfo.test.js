import React from 'react'
import { shallow } from 'enzyme'
import WorkerInfo from './WorkerInfo'

describe('WorkerInfo', () => {
  it('Should render the image and the name', () => {
    const output = shallow(<WorkerInfo imageUrl="test.jpg" firstName="Charlie" lastName="Simpson" />)
    expect(output).toMatchSnapshot()
  })
})