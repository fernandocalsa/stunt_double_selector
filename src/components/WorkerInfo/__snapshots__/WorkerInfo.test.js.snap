// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerInfo Should render the image and the name 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "border": "#000 solid 1px",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "column",
      "padding": "5px 20px",
      "width": "100%",
    }
  }
>
  <img
    alt="Charlie Simpson"
    draggable="false"
    src="test.jpg"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span
    style={
      Object {
        "fontFamily": "Tahoma, Geneva, sans-serif",
        "fontSize": 20,
      }
    }
  >
    Charlie
     
    Simpson
  </span>
</div>
`;
